### PR TITLE
Integrate Pay Modal with TransactionButton, disable Pay Modal for gasless

### PR DESCRIPTION
**Pull request by MananTank**
## PR-Codex overview
This PR integrates a Pay Modal with the TransactionButton component and allows customization. It also disables the Pay Modal based on certain configurations.

### Detailed summary
- Integrated Pay Modal with TransactionButton
- Added `payModal` prop to TransactionButton for customization
- Disabled Pay Modal based on configurations in `useSendTransaction` hook

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: packages/thirdweb/src/react/web/hooks/useSendTransaction.tsx [hunk 1]
```diff
@@ -1,5 +1,7 @@
 import { useContext, useState } from "react";
 import type { ThirdwebClient } from "../../../client/client.js";
+import { useActiveWallet } from "../../../exports/react-native.js";
```
Comment (gregfromstl) [inline packages/thirdweb/src/react/web/hooks/useSendTransaction.tsx:3]: Is this the right import (from react-native exports)?

**Review by gregfromstl — COMMENTED** [review]
Just a question on some imports otherwise lgtm

File: packages/thirdweb/src/react/web/hooks/useSendTransaction.tsx (hunk 1 above)
Comment (jnsdls) [reply packages/thirdweb/src/react/web/hooks/useSendTransaction.tsx:3]: definitely not

Comment (MananTank) [reply packages/thirdweb/src/react/web/hooks/useSendTransaction.tsx:3]: aah - auto imports 😞

Comment (MananTank) [reply packages/thirdweb/src/react/web/hooks/useSendTransaction.tsx]: Fixed now

File: packages/thirdweb/src/react/web/hooks/useSendTransaction.tsx [hunk 2]
```diff
@@ -1,8 +1,10 @@
 import { useContext, useState } from "react";
 import type { ThirdwebClient } from "../../../client/client.js";
+import type { Wallet } from "../../../exports/wallets.js";
```
Comment (gregfromstl) [inline packages/thirdweb/src/react/web/hooks/useSendTransaction.tsx:3]: Missed this one the first time I think

**Review by gregfromstl — APPROVED** [review]

File: packages/thirdweb/src/react/web/hooks/useSendTransaction.tsx (hunk 2 above)
Comment (MananTank) [reply packages/thirdweb/src/react/web/hooks/useSendTransaction.tsx:3]: let me add a biome rule for this

Comment (MananTank) [reply packages/thirdweb/src/react/web/hooks/useSendTransaction.tsx]: Hmm, Looks like Biome does not support regex in noRestrictedImports rule - I'll have to look into this. Fixing manually for now

**Pull request merged**